### PR TITLE
added tests, fixed bug where value is a comma separated list of values

### PR DESCRIPTION
**Pull request by gcase555**
Fixed a bug where a header value of comma separated items would not be matched.

- [x] Fix bug by splitting on a header value and `,`. This doesn't break things for a value without a `,` since go will just return the value anyway after doing the split even if no ,` is present.
- [x] Add unit tests to cover this edge case for checking if a value is present and not present
- [x] Add some comments for clarity and to improve readability
- [x] deploy and test this out in an environment 

**NOTE:**
I have made a separate ticket to make the header value we split on configurable and improve the error logs returned to clients. I would rather merge this pull request first to fix the blocking bug and do the improvements as a fast follow.

### CODE REVIEW

**Comment by gcase555** [comment]
FYI this should fix it @LorenBoren and @faldridge 

I will deploy this to an environment tomorrow and test it out end to end but these unit tests _should_ cover that edge case we ran into.

File: pkg/components/validate_headers_test.go [hunk 1]
```diff
@@ -14,63 +14,66 @@ func Test_incomingMatchesAllowed(t *testing.T) {
...
+			name:           "single incoming header value separated by a comma with a single allowed value",
+			allowedHeader:  map[string][]string{"Ldap-Groups": {"ops"}},
+			incomingHeader: map[string][]string{"Ldap-Groups": {"security,ops"}},
 			wantErr:        false,
 		},
 		{
 			name:           "single incoming header value with multiple allowed values",
 			allowedHeader:  defaultAllowedHeaderAndValues,
 			incomingHeader: map[string][]string{"Ldap-Groups": {"devs"}},
-			wantResult:     true,
```
Comment (gcase555) [inline pkg/components/validate_headers_test.go]: didn't need to check for this anymore now the method no longer returns a boolean and only returns an error if no match is found

**Comment by gcase555** [comment]
Change has been deployed and the edge case has been tested in an environment.
